docs/retention/README.md

```diff
@@ -0,0 +1,38 @@
+# Object Lock and Immutablity [![Slack](https://slack.min.io/slack?type=svg)](https://slack.min.io)
+
+MinIO server allows to set bucket level WORM which makes objects in the bucket immutable i.e. delete and overwrite are not allowed till stipulated time specified in the bucket's object lock configuration.
+
+## Get Started
+
+### 1. Prerequisites
+
+Install MinIO - [MinIO Quickstart Guide](https://docs.min.io/docs/minio-quickstart-guide).
+
+### 2. Run MinIO
+
+Its required to remove previously configured `worm` configuration in configuration file and `MINIO_WORM` environment variable before starting MinIO.
+
+To update the configuration, use `mc admin config get` command to get the current configuration file for the minio cluster in json format, and save it locally.
+
+```sh
+$ mc admin config get myminio/ > /tmp/myconfig
```

Comment:
We should talk about migration here i.e before upgrading MinIO proceed to first with changing the WORM configuration etc.

This document should also use AWS CLI perhaps, for now, to show how to set retention config with MinIO etc - once we have an `mc` equivalent in future we will change this document at that point.

Reply:
I think running `### 2. Run MinIO` conveys configuration change and MINIO_WORM environment variable. could you be specific to the migration?

Reply:
This document doesn't say worm is being deprecated for a newer and better approach. It also doesn't state that it should be done before upgrade

Reply:
done